### PR TITLE
feat(react-positioning): allow to configure boundaries with "PositioningRect"

### PR DESCRIPTION
**Pull request by layershifter**
## New Behavior

Floating UI supports defining boundaries as `Rect`s (https://floating-ui.com/docs/detectOverflow#boundary), this PR allows it via types.

- Exposes  `PositioningBoundary` & `PositioningRect` types in `@fluentui/react-components`
- `Boundary` type in `@fluentui/react-positioning` was marked as deprecated as it will be confusing once it's exported in `@fluentui/react-components`
- Minor rename of story files to follow the same naming format
- Adds new story

### CODE REVIEW

File: packages/react-components/react-positioning/src/types.ts [hunk 1]
```diff
@@ -51,7 +51,17 @@ export type Alignment = 'top' | 'bottom' | 'start' | 'end' | 'center';
 export type AutoSize = 'height' | 'height-always' | 'width' | 'width-always' | 'always' | boolean;
 export type NormalizedAutoSize = { applyMaxWidth: boolean; applyMaxHeight: boolean };
 
-export type Boundary = HTMLElement | Array<HTMLElement> | 'clippingParents' | 'scrollParent' | 'window';
+export type PositioningBoundary =
+  | PositioningRect
```
Comment (ling1726) [inline packages/react-components/react-positioning/src/types.ts:55]: Let's add a test for the rect boundary (popper -> floating UI upgrade still gives me trauma)

**Review by ling1726 — APPROVED** [review]

**Review by sopranopillow — APPROVED** [review]

**Review by mainframev — APPROVED** [review]

File: packages/react-components/react-positioning/src/types.ts (hunk 1 above)
Comment (layershifter) [reply packages/react-components/react-positioning/src/types.ts:55]: Added [f5fffa9](https://github.com/microsoft/fluentui/pull/33724/commits/f5fffa93191ac21f38a49d5bc80fe084bfdefb27)
